pages/course/[course]/assignment/[assignment]/submit/index.tsx

```diff
@@ -0,0 +1,104 @@
+import withProtected from "../../../../../../util/withProtected";
+import { queryParamToNumber } from "../../../../../../util/misc";
+import { GetServerSidePropsContext, NextPage } from "next";
+import Badge, { BadgeVariant } from "Components/Badge";
+import Sidebar from "Components/Sidebar";
+import { User, Assignment } from "types";
+import { createServerSupabaseClient } from "@supabase/auth-helpers-nextjs";
+import Upload from "../../../../../../Components/UploadBox";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useState } from "react";
+
+type Submission = {
+  id: string;
+  is_late: boolean;
+  score: number | null;
+  flags: string[] | null;
+  student: User;
+};
+
+type AssignmentT = Assignment & { submission: Submission[] };
+
+interface AssignmentProps {
+  id: number;
+  courseId: number;
+  assignment: AssignmentT;
+}
+
+export const getServerSideProps = (ctx: GetServerSidePropsContext) =>
+  withProtected(ctx, async () => {
+    const assignmentId = queryParamToNumber(ctx.query?.assignment);
+    const courseId = queryParamToNumber(ctx.query?.course);
+
+    const supabase = createServerSupabaseClient(ctx);
+    const assignmentData = await supabase
+      .from("assignment")
+      .select(
+        `
+        id,
+        title,
+        description,
+        is_open,
+        is_late, 
+        submission (
+          id,
+          is_late,
+          score,
+          flags,
+          student (
+            id,
+            euid,
+            given_name,
+            family_name
+          )
+        )
+        `,
+      )
+      .eq("id", assignmentId)
+      .order("created_at", { foreignTable: "submission", ascending: false })
+      .single();
+
+    return {
+      props: {
+        id: assignmentId,
+        courseId,
+        assignment: assignmentData.data,
```

Comment:
this is an optional value- make sure to handle 404 case since you're not optional chaining in the NextPage.